examples/model_selection/plot_likelihood_ratios.py

```diff
@@ -0,0 +1,322 @@
+# few samples: some folds will have no samples with the disease and therefore
+# they will output no true positives nor false negatives when used for testing.
+# Mathematically this leads to an infinite `LR+`, which should also not be
+# interpreted as the model perfectly identifying positive cases. Such event
+# leads to a higher variance of the estimated likelihood ratios, but can still
+# be interpreted as an increment of the post-test odds of having the condition.
+
+X, y = make_classification(n_samples=300, weights=[0.9], random_state=0)
+_ = cross_validate_LRs(LogisticRegression(), X, y)
+
+# %%
+# Plot class likelihood ratios
+# ============================
+#
+# The likelihood ratios are independent of the disease prevalence and can be
+# extrapolated between populations regardless of any possible class imbalance,
+# as long as the same model is applied to all of them. In the case where
+# models/diagnostic tests are updated, for instance, to evolve along with the
+# spread of a disease, one has to account for the prevalence seen during
+# training before making a comparison analysis.
+#
+# Here we train a `LogisticRegression` base model on a population with a
+# prevalence of 50%. It is then evaluated over populations with varying
+# prevalence. We use the `make_classification` function to ensure the
+# data-generating process is always the same as shown in the plots below. The
+# label `1` corresponds to the positive class "disease", whereas the label `0`
+# stands for "no-disease".
+
+import numpy as np
+import matplotlib.pyplot as plt
+from sklearn.inspection import DecisionBoundaryDisplay
+
+pos_LRs = []
+neg_LRs = []
+pos_LRs_std = []
+neg_LRs_std = []
+prevalence = []
+
+weights = np.linspace(0.1, 0.8, 6)
+weights = weights[::-1]
+
+# create base model
+X, y = make_classification(
+    n_samples=10_000,
+    n_features=2,
+    n_informative=2,
+    n_redundant=0,
+    weights=[0.5],
+    random_state=0,
+)
+clf = LogisticRegression()
+
+# fit and evaluate base model
+pos_LR_base, neg_LR_base, pos_LR_base_std, neg_LR_base_std = cross_validate_LRs(
+    clf, X, y, verbose=False
+)
+
+plt.figure(figsize=(15, 12))
+plt.subplots_adjust(hspace=0.25)
+
+for n, weight in enumerate(weights):
+
+    X_test, y_test = make_classification(
+        n_samples=10_000,
+        n_features=2,
+        n_informative=2,
+        n_redundant=0,
+        weights=[weight],
+        random_state=0,
+    )
+
+    # plot decision boundary of base model with varying prevalence
+    ax = plt.subplot(3, 2, n + 1)
+    disp = DecisionBoundaryDisplay.from_estimator(
+        clf,
+        X_test,
+        response_method="predict",
+        alpha=0.5,
+        ax=ax,
+    )
+    scatter = disp.ax_.scatter(X_test[:, 0], X_test[:, 1], c=y_test, edgecolor="k")
+    disp.ax_.set_title(f"prevalence = {y_test.mean():.2f}")
+    disp.ax_.legend(*scatter.legend_elements())
+
+    # recompute likelihood ratios for each prevalence
```

Comment:
I would use bootstraping here instead of a stratified evaluation.
With the code above, I would first define the following function for the evaluation:

```python
# %%
from collections import defaultdict


def scoring_on_bootstrap(estimator, X, y, rng, n_bootstrap=100):
    results_for_prevalence = defaultdict(list)
    for _ in range(n_bootstrap):
        bootstrap_indices = rng.choice(
            np.arange(X.shape[0]), size=X.shape[0], replace=True
        )
        for key, value in scoring(
            estimator, X[bootstrap_indices], y[bootstrap_indices]
        ).items():
            results_for_prevalence[key].append(value)
    return pd.DataFrame(results_for_prevalence)
```

And then the loop would look like this:

```python
esults = defaultdict(list)
n_bootstrap = 100
rng = np.random.default_rng(seed=0)

fig, axs = plt.subplots(3, 2, figsize=(15, 12))
for ax, (n, weight) in zip(axs.ravel(), enumerate(weights)):

    X_test, y_test = make_classification(
        **common_params_make_classification, weights=[weight]
    )

    # plot decision boundary of base model with varying prevalence
    disp = DecisionBoundaryDisplay.from_estimator(
        estimator,
        X_test,
        response_method="predict",
        alpha=0.5,
        ax=ax,
    )
    scatter = disp.ax_.scatter(X_test[:, 0], X_test[:, 1], c=y_test, edgecolor="k")
    disp.ax_.set_title(f"prevalence = {y_test.mean():.2f}")
    disp.ax_.legend(*scatter.legend_elements())

    results_for_prevalence = scoring_on_bootstrap(
        estimator, X_test, y_test, rng, n_bootstrap=n_bootstrap
    )
    results["prevalence"].append(y_test.mean())
    results["metrics"].append(results_for_prevalence.aggregate(["mean", "std"]).unstack())

results = pd.DataFrame(results["metrics"], index=results["prevalence"])
results.index.name = "prevalence"
```

The output will look like:

<img width="329" alt="image" src="https://user-images.githubusercontent.com/7454015/166905491-f11aab93-ddf3-4d31-9788-b3c30785a792.png">

That is a dataframe that you can use afterwards.